### DISCUSSION TITLE
Why do workers in Azure multiplicate the virtual users

### DISCUSSION THREAD

**Discussion by mmathoosh** [artilleryio/artillery#3356] [Artillery]
I am running tests with Playwright engine and in the yml setup I have e.g.:
config:
  target: https://xyz.abc
  phases:
      - arrivalCount: 10
        duration: 600

Why using 10 workers in run-aci (azure storage) multiplies the number of virtual users ending at 10*10=100?

My understanding was that if the load set by yml setup cannot be handled by 1 worker, I will raise workers and the problem is solved.
Meaning 10 workers -> each will take 1 virtual user and operate. (10*1 = arrivalCount: 10)

**Comment by hassy** [comment]
That's by design @mmathoosh. `--count` works as a multiplier. We'll make it clearer in the docs. Think of a worker as a separate instance of Artillery running your script. On Azure, each worker runs on a separate container instance with its own CPU and memory. Each of those workers can run some number of virtual users (this depends on CPU & memory usage of your test script).

When the desired load can't be handled by one worker, you need to scale out horizontally by adding more workers, and at the same time reducing your `arrivalRate`/`arrivalCount` values accordingly to maintain the total number.

**Reply by mmathoosh** [reply]
thanks for the reply! I appreciate it
